bip-0173.mediawiki

```diff
@@ -264,6 +267,9 @@ P2PKH addresses can be used.
 
 The following strings have a valid Bech32 checksum.
 * <tt>A12UEL5L</tt>
+* <tt>a12uel5l</tt>
+* <tt>10a06t8</tt>
```

Comment:
There is a requirement that the HRP is non-empty, so while you're technically right that this has a valid checksum, it's not testable... any decoder should fail on that input.

So I think it's maybe better to rename the section to "The following strings are valid Bech32" (and the next section to "The following string are not valid Bech32 (with reason for invalidity)", and have these two example with empty HRP in that section (with a note that technically their checksum is valid).

Reply:
I read the BIP contrary to that. My interpretation was that no such requirement that the HRP is non-empty is in the BIP. The BIP even says that the HRP's validity is application specific (which builds on top of generic Bech32). That's why this works:
```
>>> segwit_addr.bech32_create_checksum('', [])
[15, 29, 15, 26, 11, 7]
>>> segwit_addr.bech32_verify_checksum('', [] + [15, 29, 15, 26, 11, 7])
True
```
It seems odd that we'd encode something that can't be decoded. I think the reference decoder is implementing a phantom requirement right now. If we add a requirement that HRP be non-empty, I think that's OK, just other things need to be updated also. Am I making sense?

Reply:
You're right - it seems the non-empty HRP isn't mentioned in the BIP anywhere, except in the test vectors.

I prefer modifying it so that it does require that, and is uniformly implemented in decoders and encoders.

Reply:
ok, i'll update to that effect.

btw, the non-checksum data portion can be empty right now. do you want to keep that as it is now? i think so, just checking.

Reply:
Yes, sounds good. Thanks!